test/IDE/complete_storage_restriction.swift

```diff
@@ -0,0 +1,60 @@
+// RUN: %batch-code-completion
+
+struct Test {
+  var storedProperty: Int
+
+  var attrBegin: Int {
+    @#^ATTRIBUTE_BEGIN^# init(newValue) {
+      print(newValue)
+    }
+    get { 1 }
+  }
+  // ATTRIBUTE_BEGIN: Keyword/None: storageRestrictions[#Accessor Attribute#];
+
+  var bothLabels: Int {
+    @storageRestrictions(#^AFTER_PAREN^#) 
+    init(newValue) {
+    }
+    get { 1 }
+  }
+  // AFTER_PAREN: Begin completions, 2 items
+  // AFTER_PAREN-DAG: Keyword/None: initializes: [#Specify variables initialized by the accessor#];
+  // AFTER_PAREN-DAG: Keyword/None: accesses: [#Specify variables accessed by the accessor#];
+
+  var secondInitializesArgument: Int {
+    @storageRestrictions(accesses: x, #^SECOND_ACCESSES_ARGUMENT^#) 
+    init(newValue) {
+    }
+  }
+  // SECOND_ACCESSES_ARGUMENT: Begin completions, 2 items
+  // SECOND_ACCESSES_ARGUMENT-DAG: Keyword/None: initializes: [#Specify variables initialized by the accessor#];
+  // SECOND_ACCESSES_ARGUMENT-DAG: Decl[InstanceVar]/CurrNominal: storedProperty[#Int#];
+  
+
+  var secondAccessesArgument: Int {
+    @storageRestrictions(initializes: x, #^SECOND_INITIALIZES_ARGUMENT^#) 
+    init(newValue) {
+    }
+    get { 1 }
+  }
+  // SECOND_INITIALIZES_ARGUMENT: Begin completions, 2 items
+  // SECOND_INITIALIZES_ARGUMENT-DAG: Keyword/None: accesses: [#Specify variables accessed by the accessor#];
+  // SECOND_INITIALIZES_ARGUMENT-DAG: Decl[InstanceVar]/CurrNominal: storedProperty[#Int#];
+}
+
+struct TestArgument {
+  var other: Int
+
+  var otherComputed: Int { 1 }
+
+  func testFunc() {}
+
+  var argument: Int {
+    @storageRestrictions(initializes: #^FIRST_ARGUMENT^#) 
+    init(newValue) {
+    }
+    get { 1 }
+  }
+  // FIRST_ARGUMENT: Begin completions, 1 item
+  // FIRST_ARGUMENT-DAG: Decl[InstanceVar]/CurrNominal: other[#Int#];
```

Comment:
Might make sense to add one more test here that checks "accesses:" as well for completeness.